### DISCUSSION TITLE
CSV export for Multidatastreams. Observed properties and units.

### DISCUSSION THREAD

**Discussion by ndevilleBE** [FraunhoferIOSB/FROST-Server#1277] [General]
Dear, 
I make use a lot of the API and I want to push my users to use it for data download. 
Their interest is mostly triggered by multidatastreams as it allows to group all the data measured by the same sensors (e.g. temperature, salinity, conductivity,...).
As those data users are scientists and by no means IT specialists, they just want to have a link to retrieve the data. And that's where the CSV export is very handy. However, the disadvantage of MD data download is that it doesn't include the observed properties nor the units of the data in the CSV export. 
I understand the difficulties to implement that as mentioned [here](https://github.com/INSIDE-information-systems/SensorThingsAPI/blob/master/CSV-ResultFormat/CSV-ResultFormat.md)
My question is: are there work being done on that improvement? I believe it would be a great addition to the API. 
I think the option to `Encoding the unitOfMeasurements, multiObservationDataTypes, ObservedProperties and Observations/result sets as json arrays in a single CSV line` would be nice so that when user split the multidatastream data column in excel (or other software) the column header is split at the same time.
I wish to help but I'm no developer so I can't write that piece of code myself.
Many thanks

**Comment by hylkevds** [comment]
The tricky part of the problem is finding a generic solution. Adding code paths specific to MultiDatastream may work on the short term, but is hard to maintain in the long term. I think a possible solution could be to allow values for $top greater than one, and reserve columns equal to the $top value.

Currently no one is actively working on the CSV result format plugin. Features are developed either because we need them ourselves, or a customer sponsors the development through a support contract or a project specifically for such a feature.
